File: backend/src/main/java/com/thiago/gestionbodega/modules/productos/entity/Producto.java
package com.thiago.gestionbodega.modules.productos.entity;

import com.thiago.gestionbodega.common.audit.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "productos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Producto extends BaseEntity {

    @Column(name = "codigo", unique = true, length = 50)
    private String codigo;

    @Column(name = "descripcion", nullable = false, length = 200)
    private String descripcion;

    @Column(name = "stock", nullable = false, precision = 10, scale = 2)
    private BigDecimal stock;

    @Column(name = "stock_minimo", nullable = false, precision = 10, scale = 2)
    private BigDecimal stockMinimo;

    @Column(name = "es_servicio", nullable = false)
    private boolean esServicio;

    @Column(name = "usa_contometro", nullable = false)
    private boolean usaContometro;

    /** True si el producto se puede canjear con vales o puntos (regla de Karina). */
    @Column(name = "es_bazar", nullable = false)
    private boolean esBazar;

    @Column(name = "activo", nullable = false)
    private boolean activo;
}

File: lib/features/productos/domain/entities/producto.dart
import 'package:equatable/equatable.dart';

class Producto extends Equatable {
  const Producto({
    required this.id,
    required this.descripcion,
    required this.stock,
    required this.stockMinimo,
    required this.esServicio,
    required this.usaContometro,
    required this.activo,
    this.codigo,
    this.esBazar = false,
  });

  final String id;
  final String? codigo;
  final String descripcion;
  final double stock;
  final double stockMinimo;
  final bool esServicio;
  final bool usaContometro;
  final bool activo;

  /// Producto del bazar: aceptable como canje de vales y/o puntos.
  /// Definido por Karina en su respuesta del 17/jun/2026.
  final bool esBazar;

  bool get stockBajo => stock <= stockMinimo;

  @override
  List<Object?> get props => [
        id,
        codigo,
        descripcion,
        stock,
        stockMinimo,
        esServicio,
        usaContometro,
        activo,
        esBazar,
      ];
}
